packages/react-router/modules/Route.js

```diff
@@ -29,37 +25,13 @@ class Route extends React.Component {
     location: PropTypes.object
   }
 
-  static childContextTypes = {
-    router: PropTypes.object.isRequired
```

Comment:
I believe that `match` should live on `context.router` for the sake of nested `<Link>`s and `<Route>`s that need to be relative. Can we keep it there for now and just remove the subscriptions in this PR?